server/build.go

```diff
@@ -42,7 +42,17 @@ func (task *BuildTask) ID() string {
 
 	if pkg.Submodule != "" {
 		name = pkg.Submodule
+	} else {
+		pack, _ := fetchPackageInfo(pkg.Name, pkg.Version)
```

Comment:
- pls handle the error in case `pack` is nil.
- consider the "." of `exports`

Reply:
Sure,  Will fix these cases tomorrow
Fixing `pack` is nil is difficult in the `ID` method coz the method is called in many places. 
I think can refactor the logic of the build later.